### DISCUSSION TITLE
keydown event deprecated?

### DISCUSSION THREAD

**Discussion by boruxk** [quasarframework/quasar#15918] [General - Components / Directives / etc] [answered]
I would like to have key events like
@keydown.enter="cancel(item)" everywhere, espacilly in q-btn.
There are some discussions in stackoverflow, but nothing works in 2023,
Also, nothing like this in documentation, is there a workaround?

**Comment by pdanpdan** [comment, marked as answer]
Make a codepen

**Reply by boruxk** [reply]
https://codepen.io/boruxk/pen/eYPaPPO

**Reply by pdanpdan** [reply]
Looks like it's working - what is the problem?

**Reply by boruxk** [reply]
@click="cancel()" is working
@keydown.esc="cancel()" is not working
there is nothing in console on escape pressing

**Reply by suleimanas** [reply]
![Screen Shot 2023-05-31 at 09 34 04](https://github.com/quasarframework/quasar/assets/29762078/0a2236a0-3938-4def-a326-f0c1404e64f7)

Works fine

**Reply by boruxk** [reply]
this is weird, your code works, thanks!

**Reply by boruxk** [reply]
I am soorry, I see now the problem.
To make keydown working you need to click the button by mouse first and after that it's working...
I guess it's because of focusing on a button or something like that...
If you try to click ESC first, it will not work...
So is there a workaround?

**Reply by boruxk** [reply]
thanks, following was helping
https://github.com/lupas/vue-keypress

**Comment by suleimanas** [comment]
https://vuejs.org/guide/essentials/event-handling.html#key-modifiers

Great article:
https://blog.boot.dev/javascript/vue-key-events/

**Reply by boruxk** [reply]
i know this docs, its vue and not quasar,
you can make it inside "input", but not inside "q-btn"

**Reply by suleimanas** [reply]
no it works with Qbtn too
